kernel/src/fs/cgroupfs/systree_node.rs

```diff
@@ -1,5 +1,46 @@
 // SPDX-License-Identifier: MPL-2.0
 
+//! Implements the cgroup nodes for the unified cgroup hierarchy (cgroup v2).
+//!
+//! This module defines the structures for cgroup nodes (`CgroupNode`) and the cgroup
+//! root (`CgroupSystem`), integrating them into the `systree`. It handles process
+//! management within cgroups and the logic for reading and writing cgroup attributes.
+//!
+//! ## Locks and Lock Ordering
+//!
+//! To ensure thread safety during concurrent operations, this module uses several
+//! locks within the cgroup nodes. Adhering to the correct lock ordering is crucial
+//! to prevent deadlocks.
+//!
+//! ### Lock Types
+//!
+//! 1.  **Controller Lock**: Each cgroup node (including the root) has a [`Controller`]
+//!     that contains a `Mutex`. This lock protects the activation state of the sub-controllers
+//!     for its children (e.g., `memory`, `pids`).
+//!
+//! 2.  **Inner Lock**: Each non-root `CgroupNode` has an `RwMutex` that protects its
+//!     `inner` data.
+//!
+//! 3.  **Children Lock**: Each cgroup node (`CgroupNode` and `CgroupSystem`) inherits
+//!     an `RwLock` from `BranchNodeFields`. This lock protects access to and
+//!     modification of the list of child cgroup nodes.
+//!
+//! 4. **Cgroup Membership Lock**: A global `Mutex` managed by `CgroupMembership` that
+//!    serializes modifications to process cgroup memberships across the entire system.
+//!
+//! ### Locking Rules
+//!
+//! To avoid deadlocks, the following lock ordering must be strictly followed:
+//!
+//! 1.  **Parent Before Child**:
+//!     When operating on both a parent and a child node, the lock on the parent
+//!     node must be acquired before the lock on the child node.
+//!
+//! 2.  **Order Within a Single Node**:
+//!     When multiple locks are needed on the same cgroup node, they must be
+//!     acquired in this specific order:
+//!     `Controller Lock` -> `Inner Lock` -> `Children Lock`
```

Comment:
Remove a white space after "x.", e.g,
```diff
-//! 2.  **Order Within a Single Node**:
-//!     When multiple locks are needed on the same cgroup node, they must be
-//!     acquired in this specific order:
-//!     `Controller Lock` -> `Inner Lock` -> `Children Lock`
+//! 2. **Order Within a Single Node**:
+//!    When multiple locks are needed on the same cgroup node, they must be
+//!    acquired in this specific order:
+//!    `Controller Lock` -> `Inner Lock` -> `Children Lock`
```

Also cgroup types like `CgroupNode`, `CgroupSystem`, `CgroupMembership` should be replaced by something like `` [`CgroupNode`] ``.